privacysandbox/privacysandbox.go

```diff
@@ -0,0 +1,191 @@
+package privacysandbox
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+)
+
+const (
+	topicSegmentSeparator = ","
+	topicMaxLength        = 10
+)
+
+type Topic struct {
+	SegTax   int
+	SegClass string
+	SegIDs   []int
+}
+
+// ParseTopicsFromHeader parses the Sec-Browsing-Topics header data into Topics object
+func ParseTopicsFromHeader(secBrowsingTopics string) []Topic {
+	var topics []Topic
+
+	for _, seg := range strings.Split(secBrowsingTopics, topicSegmentSeparator) {
+		if topic, ok := parseTopicSegment(seg); ok {
+			topics = append(topics, topic)
+		}
+
+		if len(topics) == topicMaxLength {
+			break
+		}
+	}
+
+	return topics
+}
+
+// parseTopicSegment parses a single topic segment from the header into Topics object
+func parseTopicSegment(seg string) (Topic, bool) {
+	seg = strings.TrimSpace(seg)
+	if seg == "" || strings.HasPrefix(seg, "();p=") {
+		return Topic{}, false
+	}
+
+	segment := strings.Split(seg, ";")
+	if len(segment) != 2 {
+		return Topic{}, false
+	}
+
+	segmentsIds := strings.TrimSpace(segment[0])
+	if len(segmentsIds) < 3 || segmentsIds[0] != '(' || segmentsIds[len(segmentsIds)-1] != ')' {
+		return Topic{}, false
+	}
+
+	segtax, segclass := parseSegTaxSegClass(segment[1])
+	if segtax == 0 || segclass == "" {
+		return Topic{}, false
+	}
+
+	return Topic{
+		SegTax:   segtax,
+		SegClass: segclass,
+		SegIDs:   parseSegmentIDs(segmentsIds[1 : len(segmentsIds)-1]),
+	}, true
+}
+
+func parseSegTaxSegClass(seg string) (int, string) {
+	taxanomyModel := strings.Split(seg, ":")
+	if len(taxanomyModel) != 3 {
+		return 0, ""
+	}
+
+	// taxanomyModel[0] is v=browser_version, we don't need it
+	taxanomyVer := strings.TrimSpace(taxanomyModel[1])
+	taxanomy, err := strconv.Atoi(taxanomyVer)
+	if err != nil || taxanomy < 1 || taxanomy > 10 {
+		return 0, ""
+	}
+
+	segtax := 600 + (taxanomy - 1)
+	segclass := strings.TrimSpace(taxanomyModel[2])
+	return segtax, segclass
+}
+
+// parseSegmentIDs parses the segment ids from the header string into int array
+func parseSegmentIDs(segmentsIds string) []int {
+	var segIDs []int
+	for _, segId := range strings.Fields(segmentsIds) {
+		segId = strings.TrimSpace(segId)
+		if segid, err := strconv.Atoi(segId); err == nil && segid > 0 {
+			segIDs = append(segIDs, segid)
+		}
+	}
+
+	return segIDs
```

Comment:
Is there a way to update the naming of `segmentIds`, `segIDs`, `segId` and `segid` to be clearer about what each variable is doing?

Reply:
added some variable consistency